datamodel/high/base/dynamic_value_test.go

```diff
@@ -130,11 +127,10 @@ func TestDynamicValue_MarshalYAMLInline(t *testing.T) {
 
 	// convert node into yaml
 	bits, _ := yaml.Marshal(rend)
-	assert.Equal(t, "properties:\n    rice:\n        type: array\n        items:\n            type: string", strings.TrimSpace(string(bits)))
+	assert.Equal(t, "properties:\n    rice:\n        $ref: '#/components/schemas/rice'", strings.TrimSpace(string(bits)))
```

Comment:
@daveshanley this changes was expected and points to a difference in behavior being introduced somewhere, but I just wanted to double check what the expected result of this test should be, should it inline the component like the original value? Or is the new value correct and a bug was fixed in the process?

Reply:
okay looks like there was some special casing around maps that is not happening anymore as the ordered maps reflection wise aren't being detected as maps I will need to dig in and see what it would take to match that reflection logic with the ordered maps

Reply:
All the logic is handled by the `NodeBuilder` in the high level model package. It's what performs the reflection and decides how to handle the data. If there is a low level node available from the high level one (as in, the node was originally pulled in from a spec) then, it uses that low level node as a way to determine order. 

For new content (modified high level model), there is no low level node, so there is no way to determine the order (which is why we have our current status quo)

The `NodeBuilder` will need to learn about the ordered maps, so it can use them effectively and render out correctly. 

Some of that code in there is a bit gnarly, lots of short variable names

Reply:
Okay so for this specific test looks like the previous assertion was actually asserting a bug. This test is calling MarshalInline and therefore all the references should be inline which now happens correctly